internal/connor/eq.go

```diff
@@ -87,6 +69,27 @@ func eq(condition, data any) (bool, error) {
 	}
 }
 
+// objectsEqual returns true if the given condition and data
+// contain equal key value pairs.
+func objectsEqual(condition map[string]any, data any) (bool, error) {
+	if data == nil {
```

Comment:
nitpick: what if `condition` is also `nil`? I think it should return true.

Reply:
good catch! should be able to handle either being nil now